src/gax.ts

```diff
@@ -292,23 +342,148 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (options) {
+    // if a user provided retry AND retryRequestOptions at call time, throw an error
+    if (gaxStreamingRetries) {
+      if (
+        options.retry !== undefined &&
+        options.retryRequestOptions !== undefined
+      ) {
+        throw new Error('Only one of retry or retryRequestOptions may be set');
+      } else {
+        if (options.retryRequestOptions !== undefined) {
+          // // Retry settings
+          if (options.retryRequestOptions.objectMode) {
+            console.log(
+              'objectMode override is not supported. It is set to true internally by default in gax.'
+            );
+          }
+          if (options.retryRequestOptions.noResponseRetries) {
+            console.log(
+              'noResponseRetries override is not supported. Please specify retry codes or a function to determine retry eligibility.'
+            );
+          }
+          if (options.retryRequestOptions.currentRetryAttempt) {
+            console.log(
+              'currentRetryAttempt override is not supported. Retry attempts are tracked internally.'
+            );
+          }
+          let retryCodesOrShouldRetryFn;
+
+          if (options.retryRequestOptions.shouldRetryFn) {
+            retryCodesOrShouldRetryFn =
+              options.retryRequestOptions.shouldRetryFn;
+          } else {
+            // default to retry code 14 per AIP-194
+            retryCodesOrShouldRetryFn = [14];
+          }
+
+          //Backoff settings
+          if (
+            options.retryRequestOptions.retries !== null &&
```

Comment:
"!== null && !== undefined" can most likely be replaced with just `if (options?.retryRequestOptions?.retries)`